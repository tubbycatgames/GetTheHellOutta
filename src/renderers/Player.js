import PropTypes from "prop-types";
import React from "react";

import { Circle } from "../components";

const RADIUS = 20;
const DIAMETER = RADIUS * 2;

const Player = ({ position }) => {
  const x = position[0] - RADIUS;
  const y = position[1] - RADIUS;
  return <Circle diameter={DIAMETER} left={x} top={y} />;
};

Player.id = "Player";

Player.propTypes = {
  position: PropTypes.arrayOf(PropTypes.number).isRequired
};

export default Player;
