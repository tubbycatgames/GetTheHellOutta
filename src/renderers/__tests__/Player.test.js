import React from "react";
import renderer from "react-test-renderer";

import Player from "../Player";

describe("Player", () => {
  it("should render a player", () => {
    const player = renderer.create(<Player position={[21, 21]} />);
    expect(player.toJSON()).toMatchSnapshot();
  });
});
