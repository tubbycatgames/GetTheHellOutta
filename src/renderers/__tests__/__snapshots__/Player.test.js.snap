// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Player should render a player 1`] = `
<View
  color="#000000"
  position={
    Vector {
      "x": 21,
      "y": 21,
    }
  }
  radius={20}
  style={
    Array [
      Object {
        "backgroundColor": "#000000",
        "borderColor": "white",
        "borderRadius": 40,
        "borderWidth": 1,
        "height": 40,
        "left": 1,
        "position": "absolute",
        "top": 1,
        "width": 40,
      },
    ]
  }
/>
`;
